python/grass/jupyter/display.py

```diff
@@ -53,6 +53,28 @@ def run(self, module, **kwargs):
         else:
             raise ValueError("Module must begin with letter 'd'.")
 
+    def __getattr__(self, attr, **kwargs):
+        """Parse attribute to GRASS display module. Attribute should be in
+        the form 'd_{module}'."""
+
+        # First check to make sure format is correct
+        if attr[:2] != "d_":
+            print(attr[:2])
+            raise ValueError("Module must begin with 'd_'.")
+        # Now reformat string
+        grass_module = f"d.{attr[2:]}"
+
+        def wrapper(**kwargs):
+            # And try to run module
+            try:
+                self.run(grass_module, **kwargs)
+            except FileNotFoundError:
+                raise ModuleNotFoundError(
```

Comment:
This is more difficult to decide. Not `ModuleNotFoundError` as that's meant for Python modules, not GRASS modules. `FileNotFoundError` might be okay, but confusing. It is a failed attribute access, so `AttributeError` seem right, but this is no longer attribute access, so maybe not. Perhaps raising a new `FileNotFoundError` with a custom message. This can include the original message or use `raise ... from ...`. You could also use `shutil.which` ahead of time and then `AttributeError` would be a clear choice. This goes a little against Python principles of trying first instead of testing, but it may result in much better behavior. There is more cases than just `FileNotFoundError`, for example `PermissionError`. `shutil.which` tries to check for all these things. It worked correctly for me when only the permissions were wrong. Other than that, it should be have like `run` (e.g. with wrong parameters).

Reply:
I went with 'FileNotFoundError ' using `raise ... from ...` in my newest commit. I wasn't quite sure how to implement the `shutil.which` method and what the advantage or expected behavior would be. But, if we feel that would be a better solution, let's talk about it at our next meeting and I'll switch.